airflow/providers/amazon/aws/utils/__init__.py

```diff
@@ -66,8 +66,8 @@ def datetime_to_epoch_us(date_time: datetime) -> int:
 
 
 def get_airflow_version() -> tuple[int, ...]:
-    val = re.sub(r"(\d+\.\d+\.\d+).*", lambda x: x.group(1), version)
-    return tuple(int(x) for x in val.split("."))
+    match = re.match(r"(\d+)\.(\d+)\.(\d+)", version)
```

Comment:
Not strictly related to this but this should probably be

```suggestion
    match = re.match(r"^(\d+)\.(\d+)\.(\d+)", version)
```

Reply:
That's the difference between `re.search` (anywhere in the string), `re.match` (at the beginning) and `re.fullmatch` (both beginning and end). You do not have to add `^` at the beginning of a `re.match` pattern.